crates/wysiwyg/src/composer_model/format.rs

```diff
@@ -73,31 +73,77 @@ where
         }
     }
 
-    pub fn format(&mut self, format: InlineFormatType) -> ComposerUpdate<S> {
+    pub(crate) fn apply_pending_formats(&mut self, start: usize, end: usize) {
+        self.state
+            .toggled_format_types
+            .clone()
```

Comment:
You should be able to avoid the clone here, if format_range and unformat_range took references to a format

Reply:
Yeah actually we want to clear anyway so we can just remove first until it's empty, no need to iterate